iOS/Collection View Management /CollectionViewDataSource.swift

```diff
@@ -319,16 +319,25 @@ extension CollectionViewDataSource {
 extension CollectionViewDataSource {
 
     /// Animates the first part of the animation cycle, applies the snapshot, then finishes the animation cycle.
+    @MainActor
```

Comment:
This function must be called on the main thread because it makes changes to collectionview.